presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java

```diff
@@ -222,6 +225,13 @@ public boolean isAdoptingMergedPreference()
         }
     }
 
+    public enum TaskSpillingStrategy
+    {
+        ORDER_BY_CREATE_TIME, // When spilling is triggered, revoke tasks in order of oldest to newest
+        ORDER_BY_REVOCABLE_BYTES, // When spilling is triggered, revoke tasks by most allocated revocable memory to least allocated revocable memory
```

Comment:
Should we treat join and aggregation spill the same way? The cost of spilling them is not uniform.

With join if you spill it then the entire probe side has to be spilled incurring a 3x IO cost.

With aggregation if you spill it will have to merge in one more run, but it can continue to use memory to avoid creating more sorted runs.

In practice whether this distinction matters often???

Reply:
There are two levels of spilling strategy:

1. Prioritizing which operator to spill within a given task.
This would be easy to add on top of the existing implementation. Modifying the `VoidTraversingQueryContextVisitor` to make a list of operators and then rank them by priority would accomplish this. This refers to how we choose to spill operators within a task. So far, these implementations do not try to distinguish between operators when choosing to spill which isn't ideal. I can look into adding this in as we see fit depending on how spilling works in production for our workload as we start rolling out soon.

2. Prioritizing which task to spill in a list of currently running tasks.
As for distinguishing between revocable bytes allocated by join operator vs agg operator and using that to prioritize which tasks to spill, that would be a bit more work. Let's circle back on this if we see issues with this in during shadow.